dev/tools/controllerbuilder/pkg/codebot/registry.go

```diff
@@ -35,3 +35,16 @@ type Tool interface {
 func GetAllTools() []Tool {
 	return tools
 }
+
+func GetFilteredTools(filter Filter) []Tool {
+	filtered := make([]Tool, 0, len(tools))
+	for _, tool := range tools {
+		if filter.Exclude(tool) {
+			continue
```

Comment:
exclude only seems to do anything if the tool is included. That seems counter intuitive. 

I would expect exclude (in the absence of include) to give me everything not excluded. That matches the tool tip but does not appear to be whats happening.

If both include and and exclude are defined its not clear what the precedence should be. I would rather just error out if both are defined and keep the logic simple.

Reply:
Ahh. You reversed the logic. Exclude with "" returns false. Include with "" returns true. Thats really confusing. This section of code needs to commented to explain that.

Reply:
Right, it is to guarantee the default behavior (if both flags are not given) keeps the same (including all). updated.

Reply:
Looks like the current logic is when one tool is set to be excluded then set it to be included doesn't work? So exclude filter takes precedence over include filter? If so, could you add a unit test case to indicate this behavior?